arm-semihosting/src/lib.rs

```
@@ -146,9 +151,38 @@ pub unsafe fn syscall1(_nr: usize, _arg: usize) -> usize {
     if cfg!(all(target_arch = "arm", not(feature = "no-semihosting"))) {
         let mut nr = _nr as u32;
         let arg = _arg as u32;
+        #[cfg(arm_isa = "a64")]
+        unsafe {
+            core::arch::asm!("HLT 0xF000", inout("r0") nr, in("r1") arg, options(nostack, preserves_flags));
+        }
+
+        #[cfg(arm_isa = "a32")]
+        unsafe {
+            // We have observed some systems that accepted the HLT instruction
+            // but not SVC instruction, even though both should be equivalent.
+            // So prefer SVC but let the user pick HLT.
+            if cfg!(feature = "use-hlt") {
+                core::arch::asm!("HLT 0xF000", inout("r0") nr, in("r1") arg, options(nostack, preserves_flags));
+            } else {
+                core::arch::asm!("SVC 0x123456", inout("r0") nr, in("r1") arg, options(nostack, preserves_flags));
+            }
+        }
+
+        #[cfg(all(arm_isa = "t32", not(arm_profile = "m")))]
+        unsafe {
+            // See note about about SVC vs HLT
+            if cfg!(feature = "use-hlt") {
+                core::arch::asm!("HLT 0x3C", inout("r0") nr, in("r1") arg, options(nostack, preserves_flags));
+            } else {
+                core::arch::asm!("SVC 0xAB", inout("r0") nr, in("r1") arg, options(nostack, preserves_flags));
+            }
+        }
+
+        #[cfg(all(arm_isa = "t32", arm_profile = "m"))]
```

If none of these `cfg`s trigger, will it still work? (If not: Can we make the build break or at least loudly warn?)

---

I may be advantageous to silently drop Semihosting calls when e.g. running `cargo clippy` using the default host target. Panicking might be sub-optimal.